tests/Unit/Test_TestHelpers.cpp

```diff
@@ -134,3 +135,13 @@ SPECTRE_TEST_CASE("Unit.TestHelpers.Derivative", "[Unit]") {
     }
   }
 }
+
+SPECTRE_TEST_CASE("Unit.TestHelpers.MAKE_GENERATOR", "[Unit]") {
+  MAKE_GENERATOR(gen1);
+  MAKE_GENERATOR(gen2);
+  // This will fail randomly every 2**32 runs.  That is probably OK.
+  CHECK(gen1() != gen2());
+
+  MAKE_GENERATOR(seeded_gen, 12345);
+  CHECK(seeded_gen() == 3992670690);
```

Comment:
This will erroneously randomly succeed once every 2**32 runs if we break the macro. This is probably okay too :)